### ISSUE TITLE
num_typos defined per field

### ISSUE THREAD

**Issue by ashwinmram** [typesense/typesense#262]
## Description

I am building a hotel reservation system where each reservation record will have a unique reference/itinerary number.  It definitely makes sense to have typo tolerance for the guest name, but for the itinerary number we should have a means to configure the num_typos to 0.

## Steps to reproduce

Have a collection with a field that is unique.  Search for that field and multiple results are displayed. 

![image](https://user-images.githubusercontent.com/13768053/115712579-861f5b80-a3a7-11eb-9477-a0afa4631b5b.png)

## Expected Behavior

Ideally only 1 record should be displayed for the record that exactly matches the itinerary number search input.

## Actual Behavior

Searching for an itinerary (unique) field displays multiple results. 

## Metadata

**Typsense Version**:

0.19.0

**OS**:

Linux, Mac

**Comment by jasonbosco** [comment]
@ashwinmram @polkabrothers @CrashLaker Here's an RC build with this feature: `typesense/typesense:0.21.0.rc8`

You can now specify a comma-separated list of `num_typos` values at search time like this: 

`?num_typos=1,0,1` 

The field order should be the same as the order of fields in `query_by`.

Would love to get your feedback!

**Comment by CrashLaker** [comment]
awesome thank you @jasonbosco 

from my first tests it's still not working well.
one example i've found

```python
search_params = {
  'q': '0007',
  'query_by': 'doc',
  'num_typos': 0,
}
```
it returns
`'snippet': '....<mark>00:07:06</mark>...'`

i'm looking for an exact match on `0007`
was this expected?

**Comment by CrashLaker** [comment]
Also it doesn't seem to be working with symbols e.g. `-`

**create collection**
```python
col = 'dummy'
client.collections.create({
    'name': col,
    'fields': [
        {'name': 'id', 'type': 'string'},
        {'name': 'doc', 'type': 'string'},
    ]
})
```

**add doc**
```python
dummy_data = {
    'foo': 'bar',
    'name': 'baz',
    'serial': 'A-B',
}

client.collections[col].documents.create({
    'id': 'A',
    'doc': json.dumps(dummy_data)
})
# {'doc': '{"foo": "bar", "name": baz", "serial": "A-B"}', 'id': 'A'}
```

**search**
```
search_params = {
    'q': 'A-B',
    'query_by': 'doc',
    'num_typos': 0,
    'page': 1,
}

{'facet_counts': [],
 'found': 0,
 'hits': [],
 'out_of': 1,
 'page': 1,
 'request_params': {'collection_name': 'dummy', 'per_page', 10', 'q': 'A-B'},
 'search_time_ms': 0}
```

**Comment by jasonbosco** [comment]
@CrashLaker The two examples you pointed out are due to how Typesense handles tokenization around special characters (See: #122) and not due to typo tolerance. 

In the first example, `00:07:06` will get indexed as `000706` which is why it is returned when searching for `0007`. Similarly, `A-B` gets indexed as `AB`. We plan to support customization of special character handling in the near future as part of #122.

**Comment by ashwinmram** [comment]
This works perfectly for my use case!!  Thanks a bunch @jasonbosco  :)

Here are the before and after screenshots:

**Before**
<img width="1351" alt="Screenshot 2021-05-21 at 7 34 34 PM" src="https://user-images.githubusercontent.com/13768053/119131210-c12ab280-ba6b-11eb-95b2-638e9370491c.png">

**After**
<img width="1352" alt="Screenshot 2021-05-21 at 7 36 49 PM" src="https://user-images.githubusercontent.com/13768053/119131340-ed463380-ba6b-11eb-9cab-8c33b0281560.png">

**Comment by kishorenc** [comment]
Shipped in `0.21.0` release.
